agroal/src/main/java/org/wildfly/extension/agroal/parser/AgroalSubsystemParser_1_0.java

```diff
@@ -0,0 +1,71 @@
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.extension.agroal.parser;
+
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.PersistentResourceXMLDescription.PersistentResourceXMLBuilder;
+import org.jboss.as.controller.PersistentResourceXMLParser;
+import org.wildfly.extension.agroal.definition.AgroalSubsystemDefinition;
+import org.wildfly.extension.agroal.definition.DataSourceDefinition;
+import org.wildfly.extension.agroal.definition.DriverDefinition;
+import org.wildfly.extension.agroal.definition.XADataSourceDefinition;
+
+import static org.jboss.as.controller.PersistentResourceXMLDescription.builder;
+
+/**
+ * The subsystem parser and marshaller, that reads the model to and from it's xml persistent representation
+ *
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+public class AgroalSubsystemParser_1_0 extends PersistentResourceXMLParser {
+
+    public static final AgroalSubsystemParser_1_0 INSTANCE = new AgroalSubsystemParser_1_0();
+
+    private static final PersistentResourceXMLDescription XML_DESCRIPTION;
+
+    static {
+        PersistentResourceXMLBuilder subsystemXMLBuilder = builder( AgroalSubsystemDefinition.INSTANCE.getPathElement(), Namespace.AGROAL_1_0.getUriString() );
+
+        PersistentResourceXMLBuilder datasourceXMLBuilder = builder( DataSourceDefinition.INSTANCE.getPathElement() );
+        DataSourceDefinition.INSTANCE.getAttributes().forEach( datasourceXMLBuilder::addAttribute );
+        subsystemXMLBuilder.addChild( datasourceXMLBuilder );
+
+        PersistentResourceXMLBuilder xaDatasourceXMLBuilder = builder( XADataSourceDefinition.INSTANCE.getPathElement() );
+        XADataSourceDefinition.INSTANCE.getAttributes().forEach( xaDatasourceXMLBuilder::addAttribute );
+        subsystemXMLBuilder.addChild( xaDatasourceXMLBuilder );
+
+        PersistentResourceXMLBuilder driverXMLBuilder = builder( DriverDefinition.INSTANCE.getPathElement() );
+        driverXMLBuilder.setXmlWrapperElement( DriverDefinition.DRIVERS_ELEMENT_NAME );
+        DriverDefinition.INSTANCE.getAttributes().forEach( driverXMLBuilder::addAttribute );
+        subsystemXMLBuilder.addChild( driverXMLBuilder );
+
+        XML_DESCRIPTION = subsystemXMLBuilder.build();
```

Comment:
the whole XML builder is fluent and can be chained in a single statement (e.g. https://github.com/wildfly/wildfly/blob/f4b6b3afdea148862a28b34082c8dd215fa77607/undertow/src/main/java/org/wildfly/extension/undertow/UndertowSubsystemParser_6_0.java#L55)